Sources/protoc-gen-swift/StringUtils.swift

```diff
@@ -274,3 +274,12 @@ func stringToEscapedStringLiteral(_ s: String) -> String {
   }
   return out + "\""
 }
+
+func isValidSwiftIdentifier(_ s: String) -> Bool {
```

Comment:
If you go this route, consider precompiling a (rather large) [`NSRegularExpression`](https://developer.apple.com/reference/foundation/nsregularexpression) as a private global and matching against that. It looks like it handles the higher Unicode code points the way it should with `\U`, so the hard part is just getting everything written down in the expression.